### PR TITLE
UI to send confirmation email for unclaimed packages

### PR DESCRIPTION
**Pull request by whit537**
Part of https://github.com/gratipay/gratipay.com/pull/4305.

### CODE REVIEW

**Comment by whit537** [comment]
The model for this is the email verification workflow.

**Comment by whit537** [comment]
Oops. We'll see what Travis does with that `set_trace`. :-) Probably just time out ...

File: www/~/%username/emails/modify.json.spt [hunk 1]
```diff
@@ -36,6 +37,11 @@ elif action == 'set-primary':
     participant.update_email(address)
 elif action == 'remove':
     participant.remove_email(address)
+elif action == 'add-email-and-claim-package':
+    package_id = request.body['package_id']
+    package = Package.from_id(package_id)
```
Comment (whit537) [inline www/~/%username/emails/modify.json.spt]: What if it's `None`?

**Comment by whit537** [comment]
Sooooo ... I've been working on teaching the claim form about when there is only one address, but the implication of https://github.com/gratipay/gratipay.com/pull/4305#issuecomment-278429627 is that there will only ever be zero or one address(!).

**Comment by whit537** [comment]
Yay for talking to users? :)

**Comment by whit537** [comment]
Though I guess that's not necessarily true for all package managers, and since we're already mostly done with accounting for multiple emails (it would be work to change the schema at this point) then we may as well go for that. But on the other hand if we want to specify where the email is coming from, that will be different for each package manager.

**Comment by whit537** [comment]
On the other hand we're trying to get this done once for npm. Future expansion is out of scope until we see traction from npm.

**Comment by whit537** [comment]
Ready for review, @mattbk @nobodxbodon @JessaWitzel et al.

**Comment by whit537** [comment]
screen shot 2017-02-08 at 5 56 01 pm

------------------

screen shot 2017-02-08 at 5 56 43 pm

----------------------

screen shot 2017-02-08 at 5 57 30 pm

File: www/on/npm/%package/index.html.spt [hunk 2]
```diff
@@ -26,16 +27,43 @@ url = 'https://npmjs.com/package/' + package.name
...
+    <p>{{ _( 'Is this yours? You can claim it on Gratipay using any email address {a}on file{_a} in the maintainers field in the npm registry.'
+           , a=('<a href="https://registry.npmjs.com/' + package.name + '">')|safe
+           , _a='</a>'|safe
+            ) }}
+    {% if nemails == 0 %}
+    <p class="sorry">{{ _("Sorry, we didn't find any email addresses on file.") }}</p>
+    {% else %}
+    <input type="hidden" name="package_id" value="{{ package.id }}">
+    <ul>
+        {% for i, email in enumerate(package.emails) %}
```
Comment (nobodxbodon) [inline www/on/npm/%package/index.html.spt]: Shall we [hide the bullets](http://stackoverflow.com/questions/1027354/need-an-unordered-list-without-any-bullets) in this email list?

File: tests/py/test_www_npm_package.py [hunk 3]
```diff
@@ -4,10 +4,33 @@
...
+
+    def test_auth_gets_send_confirmation_page_from_unclaimed(self):
+        self.make_participant('bob', claimed_time='now')
+        body = self.client.GET('/on/npm/foo/', auth_as='bob').body
+        assert 'npm/foo</a> has not been claimed' in body
+        assert 'using any email address' in body
+        assert 'alice@example.com' in body
+
+    def test_auth_gets_multiple_options_if_present(self):
+        self.make_package('npm', 'bar', 'Bar', ['alice@example.com', 'alice@example.net'])
```
Comment (nobodxbodon) [inline tests/py/test_www_npm_package.py]: nit: use constant like 9c05276?

Comment (whit537) [inline tests/py/test_www_npm_package.py]: Done in 3586969370852adab37abfef8fb70db3f2ebfc66.

File: www/on/npm/%package/index.html.spt (hunk 2 above)
Comment (whit537) [inline www/on/npm/%package/index.html.spt]: Done in 455f06c67e56027839582c62ceeca8bab12b5351.

screen shot 2017-02-09 at 8 53 14 am

**Comment by whit537** [comment]
Kicked Travis, green now. We good here?

**Pull request merged**